src/inventory/beacon_inventory.rs

```diff
@@ -0,0 +1,516 @@
+                    && get_texture(15, &basic_elements) != get_texture(i + 9, &basic_elements)
+                {
+                    // if the button is pressed and not already set
+                    self.info.effect1 = this_btn_effect;
+                    if self.info.power_level == 4 {
+                        set_button_state(Active, 7, basic_elements);
+                    }
+                    set_secondary_power(get_texture(i + 9, &basic_elements), basic_elements);
+
+                    //reset previously pressed buttons
+                    for j in 0..n.min(5) {
+                        if j != i {
+                            set_button_state(Active, j + 1, basic_elements);
+                        }
+                    }
+                }
+
+                // choose effect level 2  or regen
+                if self.info.power_level == 4 && get_button_state(i + 1, &basic_elements) == Pressed
+                {
+                    if get_button_state(6, &basic_elements) == Pressed
+                        && self.info.effect2 != Some(Effect::Regeneration)
+                    {
+                        set_button_state(Active, 7, basic_elements);
+                        self.info.effect2 = Some(Effect::Regeneration);
+                    } else if get_button_state(7, &basic_elements) == Pressed
+                        && self.info.effect2 != this_btn_effect
+                    {
+                        set_button_state(Active, 6, basic_elements);
+                        self.info.effect2 = this_btn_effect;
+                    }
+                }
+
+                // make the tick active
+                if let Some(item) = self.slots.get_item(0) {
+                    if item.material == Material::GoldIngot
+                        || item.material == Material::Diamond
+                        || item.material == Material::IronIngot
+                        || item.material == Material::Emerald
+                        || item.material == Material::NetheriteIngot
+                    {
+                        set_button_state(Active, 8, basic_elements)
+                    } else {
+                        set_button_state(Inactive, 8, basic_elements)
+                    }
+                } else {
+                    set_button_state(Inactive, 8, basic_elements)
+                }
+            }
+        }
+    }
+
+    fn ty(&self) -> InventoryType {
+        InventoryType::Beacon
+    }
+}
+
+fn set_secondary_power(texture: String, basic_elements: &mut Vec<ui::ImageRef>) {
+    basic_elements.get_mut(15).unwrap().borrow_mut().texture = texture;
+    basic_elements.get_mut(15).unwrap().borrow_mut().colour.3 = 255;
+    basic_elements.get_mut(7).unwrap().borrow_mut().colour.3 = 255;
+}
+
+fn get_texture(element: usize, basic_elements: &Vec<ui::ImageRef>) -> String {
+    if let Some(el) = basic_elements.get(element) {
+        el.borrow().texture.clone()
+    } else {
+        unreachable!()
```

Comment:
why didn't you just do unwrap instead of doing a let binding and adding unreachable in the second case?